text/00110-compatibility-strategy.md

```diff
@@ -0,0 +1,364 @@
+---
+feature name: compatibility-strategy
+start date: February 2nd, 2020
+rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/111
+related issue: https://github.com/aws/aws-cdk-rfcs/issues/110
+---
+
+<!-- replace the blockquoted sections with your content -->
+# Summary
+
+There are currently two statements we advertise to our customers with regards to framework and cli compatibility:
+
+- CLI versions should always support older framework versions. That is, upgrading the CLI should never require an upgrade of the framework.
+- Framework versions might require an upgrade of the CLI. In which case, we provide instructions as to which CLI version should be used.
+
+This RFC details the necessary strategy and mechanism for ensuring that we keep those promises when releasing new versions.
+
+# Motivation
+
+There are 2 main motivations why we should put some substantial effort into this:
+
+## (1) Haven't we suffered enough?
+
+Recently, we experienced several incidents that are directly related to this domain:
+
+- https://github.com/aws/aws-cdk/issues/5986
+- https://github.com/aws/aws-cdk/issues/4897
+- https://github.com/aws/aws-cdk/issues/4294
+
+These issues were very awkward and caused both our customers and ourselves a lot of unwanted work.
```

Comment:
This statement isn't substantiated, so it feels emotional rather than objective.

Reply:
You got me there. Have you considered side gigging as a psychologist?

Reply:
That was a serious career I considered when I was getting A levels... Psychiatrist. Lawyer. Computer Scientist. Well you know which way I went :)